### PR TITLE
Auto OCP-21812 OCP-15473

### PR DESCRIPTION
**Pull request by huiran0826**
Auto OCP-21812 OCP-15473
https://openshift-qe-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/Runner-v3/56299/consoleFull
@zhaozhanqi  @anuragthehatter @weliang1 @rbbratta

### CODE REVIEW

File: features/networking/egress-ip.feature [hunk 1]
```diff
@@ -111,3 +111,69 @@ Feature: Egress IP related features
...
+    And the output should not contain:
+      |  <%= cb.valid_ip %> |
+    When I run command on the "<%= node.name%>" node's sdn pod:
+      | bash | -c | ovs-ofctl dump-flows br0 -O OpenFlow13 \| grep table=100 |
+    Then the step should succeed
+    And the output should not contain "reg0=0x"
+    """
+
+    # Remove egress cidr
+    Given as admin I successfully merge patch resource "hostsubnet/<%= node.name %>" with:
```
Comment (zhaozhanqi) [inline features/networking/egress-ip.feature]: better move this step into 'cluean-up'
```
    Given I register clean-up steps:
     """
     """
```
and you can move this after line 126.

File: features/networking/egress-ip.feature [hunk 2]
```diff
@@ -111,3 +111,69 @@ Feature: Egress IP related features
...
+  @admin
+  @destructive
+  Scenario: Should remove the egressIP from the array if it was not being used
+    Given I store a random unused IP address from the reserved range to the clipboard
+    And evaluation of `IPAddr.new("<%= cb.valid_ip %>").to_i + 1 ` is stored in the :newipint clipboard
+    And evaluation of `IPAddr.new(<%= cb.newipint %>, Socket::AF_INET).to_s` is stored in the :newip clipboard
+
+    #Patch egress cidr to the node
+    Given as admin I successfully merge patch resource "hostsubnet/<%= node.name %>" with:
+      | {"egressCIDRs": ["<%= cb.subnet_range %>"] }   |
```
Comment (zhaozhanqi) [inline features/networking/egress-ip.feature]: no need step ` the step should succeed`?

File: features/networking/egress-ip.feature [hunk 3]
```diff
@@ -111,3 +111,69 @@ Feature: Egress IP related features
...
+
+    # Patch egress IP to the project twice
+    Given I have a project
+    And as admin I successfully merge patch resource "netnamespace/<%= project.name %>" with:
+      | {"egressIPs": ["<%= cb.newip %>"]} |
```
Comment (zhaozhanqi) [inline features/networking/egress-ip.feature]: same above

File: features/networking/egress-ip.feature [hunk 4]
```diff
@@ -111,3 +111,69 @@ Feature: Egress IP related features
...
+    And as admin I successfully merge patch resource "netnamespace/<%= project.name %>" with:
+      | {"egressIPs": ["<%= cb.valid_ip %>"]} |
```
Comment (zhaozhanqi) [inline features/networking/egress-ip.feature]: same above

File: features/networking/egress-ip.feature [hunk 5]
```diff
@@ -111,3 +111,69 @@ Feature: Egress IP related features
...
+      | OPENSHIFT-FIREWALL-ALLOW  |
+
+    #check related openflow added
+    When I run command on the "<%= node.name%>" node's sdn pod:
+      | bash | -c | ovs-ofctl dump-flows br0 -O OpenFlow13 \| grep table=100 |
+    Then the step should succeed
+    And the output should contain "reg0=0x"
+
+    #Remove egress ip from namespace
+    Given as admin I successfully merge patch resource "netnamespace/<%= project.name %>" with:
```
Comment (zhaozhanqi) [inline features/networking/egress-ip.feature]: and here

File: features/networking/egress-ip.feature (hunk 1 above)
Comment (huiran0826) [reply features/networking/egress-ip.feature]: Thanks for point out this, update it.

File: features/networking/egress-ip.feature (hunk 2 above)
Comment (huiran0826) [reply features/networking/egress-ip.feature]: Actually in the step definition, it has check about success or not:
 if ! success
    raise "patch failed to apply at #{failpath}! #{@result[:response]}"
  end

I checked this kind of step in all other cases, no additional step "the step should succeed" after it.

File: features/networking/egress-ip.feature (hunk 3 above)
Comment (huiran0826) [reply features/networking/egress-ip.feature]: same as above

File: features/networking/egress-ip.feature (hunk 4 above)
Comment (huiran0826) [reply features/networking/egress-ip.feature]: same as above

File: features/networking/egress-ip.feature (hunk 5 above)
Comment (huiran0826) [reply features/networking/egress-ip.feature]: same as above

File: features/networking/egress-ip.feature (hunk 2 above)
Comment (zhaozhanqi) [reply features/networking/egress-ip.feature]: ok, then LGTM, thanks

**Pull request merged**